be/src/olap/compaction.cpp

```diff
@@ -23,6 +23,8 @@
 #include "util/trace.h"
 
 using std::vector;
+using ReadLock = std::shared_lock<std::shared_mutex>;
+using WriteLock = std::unique_lock<std::shared_mutex>;
```

Comment:
Maybe we could move these declarations to a common header file if you have replace all write lock and read lock implementations.